### ISSUE TITLE
Regex matching not working

### ISSUE THREAD

**Issue by chaoscreater** [JasonGrass/auto-extension-manager#112]
I have the following regex pattern:

```
https:\/\/login\.microsoftonline\.com\/\w+
^teams\.microsoft\.com$
```

![image](https://github.com/JasonGrass/auto-extension-manager/assets/18227319/ebae2490-21d0-4afa-a7fb-bc95c6bc67c0)

Basically, anything that is login.microsoftonline.com/* will be matched. Also, the exact URL teams.microsoft.com will also be matched. However, teams.microsoft.com/* will not be matched.

These regex have been tested and works fine on regex101.com:

![image](https://github.com/JasonGrass/auto-extension-manager/assets/18227319/5a072668-9a77-4feb-99f2-5f46fa902c19)

![image](https://github.com/JasonGrass/auto-extension-manager/assets/18227319/c604b378-9849-4382-9ae3-8cb1b702b9a8)

![image](https://github.com/JasonGrass/auto-extension-manager/assets/18227319/b5979a9a-96ec-4ea1-ab01-c89a9490949d)

![image](https://github.com/JasonGrass/auto-extension-manager/assets/18227319/941388c5-d8b5-468d-a33e-69fb2fcb3730)

But for some reason, Extension Manager just isn't doing anything when the regex condition is true.

**Comment by chaoscreater** [comment]
Here's another example:

https:\/\/teams\.microsoft\.com\/_#\/calendarv2.*

should match this URL:
https://teams.microsoft.com/_#/calendarv2

But it's not doing anything.

**Comment by JasonGrass** [comment]
you can try 

`teams\.microsoft\.com$`
>  A complete URL includes the http protocol part, so you cannot use `^`  in regular expressions here.

`https:\/\/teams.microsoft.com\/_#\/calendarv2.*`

**Comment by chaoscreater** [comment]
Thanks, I got it working.

Another question, say if I have 2 rules.

Rule A:
- https://www.google.com
- https://microsoft.com
The rule is set to Disable only when matched. This means it will enable itself when the rule doesn't match

Rule B:
- https://yahoo.com
The rule is set to Disable only when matched. This means it will enable itself when the rule doesn't match

Both rule will disable the Chrome extension named ABC.

Now, say if I browse to https://yahoo.com. Rule B should disable extension ABC, but rule A enables the extension ABC because yahoo.com is not found in either of the 2 URLs. However, I don't really want this behaviour.

If I go to yahoo.com, I expect the extension ABC to be disabled. If I leave yahoo.com, I expect the extension ABC to be enabled. I want this to only be done by the rule that is related to yahoo.com. No other rules should be interfering with disabling/enabling the extension. In other words, I want to have an isolated/sandboxed rule. Rule B should be active when yahoo.com is matched and if Rule B is active, all other rules should not kick in, until I have browsed AWAY from yahoo.com.

Perhaps there's a way to disable/enable rules based on conditions?
